modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/behavior/TableViewBehaviorBase.java

```diff
@@ -146,20 +146,20 @@ public TableViewBehaviorBase(C control) {
                 new KeyMapping(PAGE_UP, e -> scrollUp()),
                 new KeyMapping(PAGE_DOWN, e -> scrollDown()),
 
-                new KeyMapping(LEFT, e -> selectLeftCell()),
-                new KeyMapping(KP_LEFT, e -> selectLeftCell()),
-                new KeyMapping(RIGHT, e -> selectRightCell()),
-                new KeyMapping(KP_RIGHT, e -> selectRightCell()),
+                new KeyMapping(LEFT, e -> { if(isRTL()) selectRightCell(); else selectLeftCell(); }),
+                new KeyMapping(KP_LEFT,e -> { if(isRTL()) selectRightCell(); else selectLeftCell(); }),
```

Comment:
OK, I checked into this a little more closely and I can't think if a cleaner way to do it. What I originally suggested (using a ternary operator) doesn't work. And the other thing I was going to suggest, that of creating new methods like `selectForwardCell` (kind of like you did in the test) isn't worth the effort, since each is only repeated twice and it would likely result in code that isn't any easier to understand or maintain. So I think what you've done is fine.

Btw, I guess you didn't mean to remove the space after the `,` here?